packages/proxy/lib/http/index.ts

```diff
@@ -138,7 +138,32 @@ export function _runStage (type: HttpStages, ctx: any, onError) {
         .value()
       }
 
+      function _onError (error: Error) {
+        ctx.debug('Error in middleware %o', { middlewareName, error })
+
+        if (type === HttpStages.Error) {
+          return
+        }
+
+        res?.off('close', onClose)
+        _end(onError(error))
+      }
+
+      function onClose () {
+        if (!res?.writableFinished) {
+          _onError(new Error('Socket closed before finished writing response.'))
+        }
+      }
+
+      // If we are in the middle of the response phase we want to listen for the on close message and abort responding and instead send an error.
+      // If the response is closed before the middleware completes, it implies the that request was canceled by the browser.
+      // The request phase is handled elsewhere because we always want the request phase to complete before erroring on canceled.
```

Comment:
> we always want the request phase to complete before erroring on canceled.

But we don't want the request phase to complete before erroring... for instance `SendRequestOutgoing` probably shouldn't run if the request is canceled. Am I misinterpreting this?

Reply:
Specifically I need the pre-request to be sent to the driver before we throw the error, otherwise the error doesn't have a request to relate to and is lost.

Reply:
Gotcha, so `CorrelateBrowserPreRequest` and `SendToDriver` are desirable. I guess this isn't the biggest deal, without Cypress's proxy, the request would be sent outgoing if aborted anyways.

Reply:
yeah, i think we could be more efficient by not sending the outgoing request but it's a race condition for sure.